File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const runningSum = (nums) => {
	let result = [nums[0]];
	for(let i = 1; i < nums.length; i++) {
		result.push(nums[i] + result[i - 1]);
	}
	return result;
}

File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* runningSum(int* nums, int numsSize, int* returnSize){
	int *result = malloc(sizeof(int) * numsSize);
	int sum = 0;

	for(int i = 0; i < numsSize; i++) {
		sum += nums[i];
		result[i] = sum;
	}
	*returnSize = numsSize;
	return result;
}
